modules/seqexec/server/src/main/scala/seqexec/server/gems/GemsController.scala

```diff
@@ -3,17 +3,20 @@
 
 package seqexec.server.gems
 
-import seqexec.server.tcs.Gaos.{PauseCondition, ResumeCondition}
+import seqexec.server.gems.Gems.GemsGuiderStatus
+import seqexec.server.tcs.Gaos.{PauseCondition, PauseResume, ResumeCondition}
 import squants.Time
 
 trait GemsController[F[_]] {
   import GemsController._
 
-  def pause(reasons: Set[PauseCondition]): F[Unit]
-  def resume(reasons: Set[ResumeCondition], config: GemsConfig): F[Unit]
+  def pauseResume(config: GemsConfig, pauseReasons: Set[PauseCondition],
+                  resumeReasons: Set[ResumeCondition]): F[PauseResume[F]]
```

Comment:
This follows the implementation of Altair. The pause and resume procedures are calculated in the same function. In this way, the resume can depend on what is done in the pause.